### PR TITLE
Closes #2593: FxA automatic sign-in integration 

### LINKED ISSUE #2593: FNX3-16435 ⁃ FxA Auto Sign-in
### Why/User Benefit/User Problem
As a user I want the option for Fenix to automatically sign me in my Firefox account if I am currently signed in FxA on Fennec on the same device. 

### Acceptance Criteria
-I see a prompt during Fenix onboarding, asking me if I want to sign into my Firefox Account (if I am already signed into FxA on Fennec on my device)
-I can tap on the prompt, and be automatically signed into FxA 
-Once signed in, I will be taken to the Sync page. 

### What / Requirements
-UX Designs ==> https://mozilla.invisionapp.com/share/XBRNF6MJVUY
~~Application Services dependency (Verify the a~~s PR for session token migration works with the a-c PR for account state machine, review, and merge)

### PR DESCRIPTION
**Pull request by grigoryk**
Depends on https://github.com/mozilla-mobile/android-components/pull/3905 and on https://github.com/mozilla-mobile/android-components/pull/3907.

### CODE REVIEW

**Comment by grigoryk** [comment]
To test this locally, here are the steps you need to take:
- build Fennec locally, disabling `isTrustedProvider` check in `AuthStateProvider.java`
- figure out certificate fingerprint of your local Fennec build. In your local mozilla-central checkout, run:
  - `cd objdir-frontend/gradle/build/mobile/android/app/outputs/apk/withoutGeckoBinaries/debug/`
  - `unzip app-withoutGeckoBinaries-debug.apk`
  - `keytool -printcert -file META-INF/CERT.RSA | grep -i sha256: | cut -d ' ' -f 3 | sed 's/://g' | tr '[:upper:]' '[:lower:]'`
  - That last command should output something that looks like `a105512ff3eb320325d558fca0c7ecf92c78bdb7e71efdbf6c93f60ee404af52`
- publish android-components locally, adding your local fennec package name and fingerprint to the whitelist in `AccountSharing.kt`
  - e.g. `"org.mozilla.fennec_grisha" to "a105512ff3eb320325d558fca0c7ecf92c78bdb7e71efdbf6c93f60ee404af52"`
- build Fenix against a locally published android-components.
- sign-into FxA on Fennec. Launch Fenix, observe the onboarding goodness.
- ... profit!

**Comment by grigoryk** [comment]
I've removed the "stay signed out" button, and the UI feels _much_ better. The state management code could be much simpler as well, as a result.

Given the ephemeral nature of the onboarding UI, I'd say we just drop that button. cc @apbitner

**Comment by apbitner** [comment]
@grigoryk I'm ok with that given the ephemeral nature as you noted

**Comment by grigoryk** [comment]
This is blocked on A-C using an A-S release with the following fix in it: https://github.com/mozilla/application-services/issues/1434

Otherwise, any failures during sign-in will result in us getting into an almost non-recoverable state (only recoverable by restart)

**Comment by ekager** [comment]
Added a do not land label, please remove when AC is updated with that fix! :)

**Comment by grigoryk** [comment]
This should be fine to land once https://github.com/mozilla-mobile/fenix/pull/4391 lands.

**Comment by grigoryk** [comment]
This is unblocked!

File: app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/OnboardingManualSignInViewHolder.kt [hunk 1]
```diff
@@ -28,28 +28,20 @@ class OnboardingFirefoxAccountViewHolder(private val view: View) : RecyclerView.
...
 
-    private fun updateButtonVisibility(autoSignedIn: Boolean) {
-        view.turn_on_sync_button.visibility = if (autoSignedIn) View.GONE else View.VISIBLE
-        view.stay_signed_in_button.visibility = if (autoSignedIn) View.VISIBLE else View.GONE
-        view.sign_out_button.visibility = if (autoSignedIn) View.VISIBLE else View.GONE
-    }
-
-    private fun updateHeaderText(autoSignedIn: Boolean) {
-        val icon = if (autoSignedIn) avatarAnonymousDrawable else firefoxAccountsDrawable
+    // TODO move this into the layout?
```
Comment (ekager) [inline app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/OnboardingManualSignInViewHolder.kt]: Which part - the icon? If it's always going to be this icon I think that makes sense

Comment (grigoryk) [reply app/src/main/java/org/mozilla/fenix/home/sessioncontrol/viewholders/onboarding/OnboardingManualSignInViewHolder.kt]: Right, we're not switching icons around anymore. But at the time I wasn't up to the task of adding an icon to the layout code myself :) I think I tried for a minute or two, and quickly gave up @ekager . I shall attempt this again!

**Pull request merged**